### ISSUE TITLE
Docker Desktop 3.5.2 Causes a host of newly introduced issues on Mac OS 11.2.3

### ISSUE THREAD

**Issue by reskin89** [docker/for-mac#5882]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 27764E49-E68F-4274-B7FD-95843D078899/20210727142136

### Expected behavior
When opening a shell in a local container, any files mounted via volume (like ssh keys) should have their password prompts hidden if they require one (such as for an ssh key that's password protected).

While in the shell of a local container, tab completion should work

While running a process in the shell of a local container, ctrl/cmd+c should terminate such process

Arrow keys while in the shell of a local container should operate as intended

### Actual behavior

> When opening a shell in a local container, any files mounted via volume (like ssh keys) should have their password prompts hidden if they require one (such as for an ssh key that's password protected).

When the password prompt arises, I can see any password in plain text that is typed

> While in the shell of a local container, tab completion should work

Tab inserts 3 spaces for some reason, and causes the cursor to become misaligned with everything

> While running a process in the shell of a local container, ctrl/cmd+c should terminate such process

It now causes a container to hang and eventually send a sigterm to the main system to kill the container

> Arrow keys while in the shell of a local container should operate as intended

They now insert command characters such as `[A`

None of these issues existed in the previous version I was on, and I am using the some container image as before.  After upgrading to 3.5.2, uninstalling, and reinstalling as well as many reboots, these issues still exist.

### Information

  - macOS Version: 11.2.3
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 3.5.2.18

### Steps to reproduce the behavior
```
docker run -it alpine:latest /bin/sh
sleep 3000

cmd+c

ls /etc/ <tab>
```

**Comment by stevebutler2210** [comment]
>Arrow keys while in the shell of a local container should operate as intended

>They now insert command characters such as [A

I'm experiencing this one, too, running macOS Version 10.15.6.

Interestingly, this only manifests for me when running a container via `docker compose` / `docker-compose`. E.g. running `docker-compose run --rm my_image bash` and then using the arrow keys inserts command characters, but running `docker run -it my_container bash` and doing the same behaves as I would expect.

**Comment by noefroidevaux** [comment]
I'm experiencing the same behaviour since the update to the version 3.5.2 (mac OS 11.5.1, iTerm2, zsh). Impossible to use arrow keys or `ctrl-c` in a container.

**Comment by ADavisDon** [comment]
I am also experiencing all the same behavior. Doing an uninstall and wiping out all Docker data before re-installing did not resolve the issue. This occurs regardless if `docker compose` or `docker-compose` is used.

**Comment by mpentecost** [comment]
I'm also experiencing this, running MacOS 10.15.7.  This unfortunately breaks using custom shells/REPL dev environments built to run from inside a container.

**Comment by jonkiparsky** [comment]
Also experiencing this. Have rolled back to 3.5.1 pending a fix. 

It would be very useful to know if there is any intention of fixing this issue. If so, my team can pin to 3.5.1 and wait. If not, I'd prefer to figure out a solution to make our existing tooling work with the New Normal.

**Comment by zaratan** [comment]
This is solved with `3.6.0` for me.

**Comment by noefroidevaux** [comment]
> This is solved with `3.6.0` for me.

On my side too.

**Comment by ADavisDon** [comment]
Confirmed  fixed on 3.6.0
